bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1258,14 +1251,31 @@ public void setEnabled(ThingUID thingUID, boolean enabled) {
         }
     }
 
+    private void persistThingEnableStatus(ThingUID thingUID, boolean enabled) {
+        if (storage != null) {
```

Comment:
I guess you actually mean `storage == null`. And I would also suggest to return in this case so we don't get a NPE later.

Reply:
Yes, correct, it should be `storage==null` Do we want to just return? This way the status of the, if the thing itself is not null will be changed but not persisted.

Reply:
It was just like this before. I suggest to keep this behaviour and return.